### PR TITLE
Move non-generic apiserver code out of the generic packages

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: cmd/kube-apiserver/app/options/validation.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2014 The Kubernetes Authors.
+Copyright 2016 The Kubernetes Authors.
```
Comment (deads2k) [inline cmd/kube-apiserver/app/options/validation.go]: date ought not change.

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -123,7 +125,7 @@ func Run(s *options.ServerRunOptions) error {
 
 	// Setup tunneler if needed
 	var tunneler genericapiserver.Tunneler
-	var proxyDialerFn apiserver.ProxyDialerFunc
+	var proxyDialerFn func(network, addr string) (net.Conn, error)
```
Comment (deads2k) [inline cmd/kube-apiserver/app/server.go]: This looks like a move in the wrong direction.  There ought to be *some* type we can use.

Comment (deads2k) [inline cmd/kube-apiserver/app/server.go]: > It's only used here.

I'd still prefer a type for it.

**Comment by deads2k** [comment]
I'd like to keep a type for the function, otherwise, lgtm.  I don't need another look.  Could use some squashing.

File: cmd/kube-apiserver/app/server.go (hunk 2 above)
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: In fact, the func type is used at several places, the alias isn't. Will factor that out.

**Comment by sttts** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**